contracts/libraries/VaultLifecycleTreasury.sol

```diff
@@ -0,0 +1,796 @@
+                vaultState.totalPending,
+                params.performanceFee,
+                params.managementFee
+            );
+        }
+
+        // Take into account the fee
+        // so we can calculate the newPricePerShare
+        currentBalance = currentBalance.sub(totalVaultFee);
+
+        {
+            newPricePerShare = ShareMath.pricePerShare(
+                params.currentShareSupply,
+                currentBalance,
+                pendingAmount,
+                params.decimals
+            );
+
+            // After closing the short, if the options expire in-the-money
+            // vault pricePerShare would go down because vault's asset balance decreased.
+            // This ensures that the newly-minted shares do not take on the loss.
+            mintShares = ShareMath.assetToShares(
+                pendingAmount,
+                newPricePerShare,
+                params.decimals
+            );
+
+            uint256 newSupply = params.currentShareSupply.add(mintShares);
+
+            queuedWithdrawAmount = newSupply > 0
+                ? ShareMath.sharesToAsset(
+                    queuedWithdrawShares,
+                    newPricePerShare,
+                    params.decimals
+                )
+                : 0;
+        }
+
+        return (
+            currentBalance.sub(queuedWithdrawAmount), // new locked balance subtracts the queued withdrawals
+            queuedWithdrawAmount,
+            newPricePerShare,
+            mintShares,
+            performanceFeeInAsset,
+            totalVaultFee
+        );
+    }
+
+    /**
+     * @notice Creates the actual Opyn short position by depositing collateral and minting otokens
+     * @param gammaController is the address of the opyn controller contract
+     * @param marginPool is the address of the opyn margin contract which holds the collateral
+     * @param oTokenAddress is the address of the otoken to mint
+     * @param depositAmount is the amount of collateral to deposit
+     * @return the otoken mint amount
+     */
+    function createShort(
+        address gammaController,
+        address marginPool,
+        address oTokenAddress,
+        uint256 depositAmount
+    ) external returns (uint256) {
+        IController controller = IController(gammaController);
+        uint256 newVaultID =
+            (controller.getAccountVaultCounter(address(this))).add(1);
+
+        // An otoken's collateralAsset is the vault's `asset`
+        // So in the context of performing Opyn short operations we call them collateralAsset
+        IOtoken oToken = IOtoken(oTokenAddress);
+        address collateralAsset = oToken.collateralAsset();
+
+        uint256 collateralDecimals =
+            uint256(IERC20Detailed(collateralAsset).decimals());
+        uint256 mintAmount;
+
+        if (oToken.isPut()) {
+            // For minting puts, there will be instances where the full depositAmount will not be used for minting.
```

Comment:
Please encapsulate hacky code into a function, easier to maintain.

Reply:
The VaultLifecycleTreasury is actually a copy of the existing VaultLifecycle. Will be making modifications to some functions, but for existing ones, I think I should keep them as is. What do you think?